app/src/main/java/com/expidev/gcmapp/MeasurementsActivity.java

```diff
@@ -67,8 +86,34 @@ protected void onStart()
 
         currentPeriod = preferences.getString("currentPeriod", null);
         setupBroadcastReceivers();
-        MinistriesService.retrieveMinistries(this);
+        startLoaders();
+    }
+
+    private void startLoaders()
+    {
+        final LoaderManager manager = this.getSupportLoaderManager();
+
+        manager.initLoader(LOADER_CURRENT_MINISTRY, null, ministryLoaderCallbacks);
+    }
+
+    @Override
+    protected void onPostResume()
+    {
+        super.onPostResume();
+        Log.i(TAG, "Resuming");
+        chosenMcc = preferences.getString("chosen_mcc", null);
+    }
+
+    private void restartMeasurementLoader(String ministryId, String mcc, String period)
+    {
+        final LoaderManager manager = this.getSupportLoaderManager();
 
+        Bundle args = new Bundle(3);
+        args.putString("ministryId", ministryId);
+        args.putString("mcc", mcc);
+        args.putString("period", period);
+
+        manager.restartLoader(LOADER_MEASUREMENTS, args, measurementsLoaderCallbacks);
     }
 
     private void setupBroadcastReceivers()
```

Comment:
not really necessary anymore. All it's doing is logging processing state, which could actually be done from the MeasurementsService itself.